pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ParserProxyHandler.java

```diff
@@ -0,0 +1,180 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pulsar.proxy.server;
+
+
+import avro.shaded.com.google.common.collect.Lists;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.buffer.Unpooled;
+import io.netty.buffer.CompositeByteBuf;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import org.apache.pulsar.common.api.proto.PulsarApi;
+import org.apache.pulsar.common.api.raw.MessageParser;
+import org.apache.pulsar.common.api.raw.RawMessage;
+import org.apache.pulsar.common.naming.TopicName;
+import org.apache.pulsar.common.util.protobuf.ByteBufCodedInputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.Map;
+
+
+public class ParserProxyHandler extends ChannelInboundHandlerAdapter {
+
+
+    private Channel channel;
+    //inbound
+    protected static final String frontendConn = "frontendconn";
+    //outbound
+    protected static final String backendConn = "backendconn";
+
+    private String connType;
+
+
+    //producerid+channelid as key
+    //or consumerid+channelid as key
+    private static Map<String, String> producerHashMap = new ConcurrentHashMap<>();
+    private static Map<String, String> consumerHashMap = new ConcurrentHashMap<>();
+
+    public ParserProxyHandler(Channel channel, String type){
+        this.channel = channel;
+        this.connType=type;
+    }
+
+    private void logging (Channel conn,PulsarApi.BaseCommand.Type cmdtype,String info,List<RawMessage> messages) throws Exception{
+
+        if (messages !=null){
+            // lag
+            for (int i=0;i <messages.size();i++){
+                info = info + "["+(System.currentTimeMillis() - messages.get(i).getPublishTime())+"] "+new String(ByteBufUtil.getBytes((messages.get(i)).getData()),"UTF8");
+            }
+        }
+
+        switch (this.connType){
+            case ParserProxyHandler.frontendConn:
+                log.info("conn:{} cmd:{} msg:{}","["+ conn.remoteAddress().toString()+""+conn.localAddress()+channel.localAddress()+channel.remoteAddress()+"]",cmdtype,info);
```

Comment:
I see `conn.localAddress()` is used twice in this logging statement. is that done intentionally?

```suggestion
                log.info("conn:{} cmd:{} msg:{}", "[" + conn.remoteAddress().toString() + conn.localAddress() + channel.localAddress() + channel.remoteAddress() +"]", cmdtype,info);
```

Reply:
good catch, actually I was trying to add the feautre(whitelist for client as well as broker side), this need get broker ip/port from inbound channel, but right now it would be in next PR. I will update this.